filters/all.non_parser_extensions.test.js

```diff
@@ -0,0 +1,39 @@
+const test = require("ava");
+const {nonParserExtensions} = require("./all");
+
+let undef;
+
+test.beforeEach(t => t.context.doc = {});
+
+test("returns an empty Map when schema is falsy", t => {
+  const result = nonParserExtensions(undef);
+  t.is(result.constructor.name, "Map");
+  t.is(result.size, 0);
+});
+
+test("returns an empty Map when schema has no extensions", t => {
+  const result = nonParserExtensions(t.context.doc);
+  t.is(result.constructor.name, "Map");
+  t.is(result.size, 0);
+});
+
+test("returns an empty Map when schema extensions is not a function", t => {
+  t.context.doc.extensions = [];
+  const result = nonParserExtensions(t.context.doc);
+  t.is(result.constructor.name, "Map");
+  t.is(result.size, 0);
+});
+
+test("returns a Map with the truthy schema extensions that do not start with x-parser-", t => {
+  t.context.doc.extensions = function() {
```

Comment:
not super important, but I just noticed it so writing about it, could be arrow function here